defaulthasher/hasher_test.go

```diff
@@ -0,0 +1,107 @@
+package defaulthasher
```

Comment:
It would be a good idea to add a bulk test like https://github.com/lazyledger/smt/blob/optimised_compute/bulk_test.go for building the tree and testing proofs. This can pick-up unexpected edge cases.